mypy/semanal.py

```diff
@@ -803,6 +813,21 @@ def analyze_overloaded_func_def(self, defn: OverloadedFuncDef) -> None:
         # We know this is an overload def. Infer properties and perform some checks.
         self.process_final_in_overload(defn)
         self.process_static_or_class_method_in_overload(defn)
+        if defn.impl:
+            self.process_overload_impl(defn)
+
+    def process_overload_impl(self, defn: OverloadedFuncDef) -> None:
+        """Set flags for an overload implementation.
+
+        Currently, this checks for a trivial body in protocols classes,
+        where it makes the method implicitly abstract.
+        """
+        assert defn.impl is not None
```

Comment:
nit: I think that it is better to use `if defn.impl is None: return` here.

And remove this check: https://github.com/python/mypy/pull/12118/files#diff-f782aad4853edceb67a88efe43d2869579d075a77b8e8dd2add91e818a9fa164R816

This way we won't have accidental `AssertionError` if someone uses this method incorrectly.

Reply:
That's of course fine by me, but note that I just moved this function and didn't modify it in any way.

Reply:
Sorry, I was confused about which function you were talking about. This is not the function I moved. I fixed it.